### PR TITLE
Add a validator that ensures one metadata is between min and max values

### PR DESCRIPTION
**Pull request by fmareuil**
I'm not sure if it's not already done but I have written a class to add a validator that ensures one metadata is between min and max values. Typically it can be used to ensure that a fasta file has the right number of sequences.
```
<param name="file_fasta" type="data" format="fasta"  label=">Reference (Fasta)">
    <validator type="dataset_metadata_in_range" max="1" metadata_name="sequences"/>
</param>
```

### CODE REVIEW

**Comment by martenson** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
Thanks a lot @fmareuil, this looks very good and useful. I've added a test case and some minor changes in https://github.com/fmareuil/galaxy/pull/1, if you merge this I think this is ready to go.

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
@galaxybot test py3

**Comment by fmareuil** [comment]
Thanks a lot @mvdbeek

File: test/functional/tools/validation_metadata_in_range.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<tool id="validation_metadata_in_range" name="validation_metadata_in_range" profile="19.05" version="0.1">
+  <command>
+    echo "Hello World" > out1;
+  </command>
+  <inputs>
+    <param name="input1" type="data" format="fasta" label="fasta file with more than 10 sequences">
```
Comment (martenson) [inline test/functional/tools/validation_metadata_in_range.xml]: should be "more than one"?

Comment (mvdbeek) [reply test/functional/tools/validation_metadata_in_range.xml]: Ah thanks, and I also forgot to list this in another place in the xsd.

Comment (mvdbeek) [reply test/functional/tools/validation_metadata_in_range.xml]: https://github.com/fmareuil/galaxy/pull/2

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
@galaxybot test py3

**Review by mvdbeek — APPROVED** [review]
Awesome, thanks a lot @fmareuil.
I'm surprised that github says it's your first contribution!

**Comment by martenson** [comment]
thanks @fmareuil !

**Comment by fmareuil** [comment]
Thanks a lot @mvdbeek and @martenson 
Yes It's my first contribution, I'm so happy about it :)

**Pull request merged**